### PR TITLE
Add thread tests

### PR DESCRIPTION
**Pull request by bennypk1**
*Added all the tests for AddThreadInteractor
* Success test not yet implemented

### CODE REVIEW

File: src/test/java/use_case/add_thread/AddThreadInteractorTest.java [hunk 1]
```diff
@@ -0,0 +1,369 @@
...
+            }
+
+            @Override
+            public void prepareFailView(String errorMessage)
+            {
+                fail("Failure is unexpected here");
+            }
+
+            @Override
+            public void switchToThreadsView()
```
Comment (Benjamin-Uoft) [inline src/test/java/use_case/add_thread/AddThreadInteractorTest.java:76]: -  Not certain if you need to test the switch to threads view method - maybe we can leave it empty if you still have to define the method here.
-

**Review by Benjamin-Uoft — APPROVED** [review]
Overall, the tests look good. We can benefit from having a few more cases to test, like a valid input data with only one user, or to test for special characters in the name. We can also probably reuse the same presenter for each test by having a switch block within the logic(hard coded based on what the input data was and what the message is supposed to be) and it would reduce the amount of code significantly.
